### DISCUSSION TITLE
SCRIPT command to get current "Set" and "Remain" value of PULSETIME

### DISCUSSION THREAD

**Discussion by Mario-75** [arendst/Tasmota#18454] [Ideas]
I would like to request to have access to the “Set” and “Remaining” value of the “Pulsetime” command within the Scripting Language.
If I start a Relay (in my example Relay2) with active PulseTime I would like to get the remaining time displayed.
The only solution I found so far is to send:
+>PulseTime2
This send an MQTT RESULT with the PulseTime Set and Remaining e.g.:
…/RESULT = {"PulseTime2":{"Set":160,"Remaining":158}}
This can be subscribed within the SCRIPT e.g.:
+>subscribe TREMAIN, stat/%topic%/RESULT, PulseTime2.Remaining
This has some delay when sending MQTT and receiving again. 
Thanks
Mario
I would like to request to have access to the Set and Remain value of the Pulsetime command within the Scripting Langueage.

**Comment by gemu2015** [comment]
you do not need to subscribe.

simply fetch the result in section >E, no delay at all

```
>E
remtime=PulseTime1#Remaining
```

**Reply by Sfre12** [reply]
Hey gemu2015
I tried this but got no response from Pulsetime in Section >E
Only if I request pulsetime every second it ist updating the variable.
Is there any other possibility?
best regards
Stephan

```
>D
; für dieses Beispielscript wird vorausgesetzt,
; das 4 Relais in der Vorlage aktiv sind.

restzeit1=0
restzeit2=0
restzeit3=0
restzeit4=0
r1=0
r2=0
r3=0
r4=0
>B
->Backlog Pulsetime1 120;Pulsetime2 120;Pulsetime3 120;Pulsetime4 120;
>BS
->Backlog Pulsetime1 120;Pulsetime2 120;Pulsetime3 120;Pulsetime4 120;

>E
restzeit1=PulseTime1#Remaining
restzeit2=PulseTime2#Remaining
restzeit3=PulseTime3#Remaining
restzeit4=PulseTime4#Remaining

>F
r1=pwr[1]
r2=pwr[2]
r3=pwr[3]
r4=pwr[4]

>S

Wenn eines der Relais mit Pulsetime aktiv ist, dann wird diese angefordert
if pwr[1]==1 
then
+>pulsetime1
endif

if pwr[2]==1
then
+>pulsetime2
endif

if pwr[3]==1
then
+>pulsetime3
endif

if pwr[4]==1
then
+>pulsetime4
endif

>W
; Hier wird das Web UI modifiziert

restzeit1: %restzeit1%
restzeit2: %restzeit2%
restzeit3: %restzeit3%
restzeit4: %restzeit4%

r1-2-3-4:%0r1%-%0r2%-%0r3%-%0r4%
```

**Reply by sfromis** [reply]
No, matching on JSON payloads is only available when Tasmota is producing those payloads, like by issuing the PulseTime command.

**Reply by Sfre12** [reply]
OK, that means I have to do it in my script exactly as it is

**Comment by Mario-75** [comment]
Thanks, that works fine for me. Actually, I tried it in the >S section before, but that doesn’t work.
Just to give a bit more explanation to others. I’m using it to turn off Relay2 after a given number of seconds. I send the command by MQTT. For example 30 Minutes: `PulseTime2 1900` 1900 (60*30+100). If I turn Relay2 ON it will turn Off after 1800 Seconds.
The script needs to send PulseTime regularely and the >E section grabs that result.
(I check if the Relay2 is ON `pwr[2]==1` otherwise it does not make sense to grab the remaining time.)
Code example:
```
>D
remtime=0

>S
if upsec%10=0
and pwr[2]==1
then
+>PulseTime2
endif

>E
remtime=PulseTime2#Remaining
```

**Comment by DC-555** [comment]
I'm having a similar problem, but not
works the same for me.
I want to collect the value "done" :

09:39:19.755 MQT: stat/Connect_V2_F870E4/RESULT = {"Sendmail":"Done"}

`>B
+>subscribe Sendmail,stat/%topic%/RESULT
>E
SM=Sendmail`

I tried so much with Subscribe, so much without but I didn't get the value in SM="".

**Reply by gemu2015** [reply]
you do not need subscribe!
you had a typo in Sendmail`  (extra char at the end)
just put this in >E 

```
>E
SM=Sendmail
```

**Reply by DC-555** [reply]
No, that character is why I tried to put it in code mode here on github (ctrl+E).
It doesn't work without signing that simple way, signing I managed to fix it like this:
>B
+>subscribe SM,stat/%topic%/RESULT,Sendmail

That's how it works.

**Reply by gemu2015** [reply]
it definitely works without subscribe, i just verified it.

**Reply by DC-555** [reply]
Can you share the script for me? please?

**Reply by gemu2015** [reply]
I guess I know what you did wrong
You must enable recursion with the send mail cmd
Use +> instead of ->

**Reply by DC-555** [reply]
>S
=>sendmail [smtp.gmail.com:465...
I'll try to change it to:

>D
+>sendmail [smtp.gmail.com:465...

**Reply by DC-555** [reply]
Yes, now it works without signing up.

SN=Sendmail
